src/send/mod.rs

```diff
@@ -0,0 +1,170 @@
+pub mod v2;
+
+use std::io::Cursor;
+use std::str::FromStr;
+use std::sync::Arc;
+
+pub use payjoin::send::RequestBuilder as PdkRequestBuilder;
+
+use crate::error::PayjoinError;
+use crate::send::v2::ContextV2;
+use crate::types::Request;
+use crate::uri::Uri;
+
+///Builder for sender-side payjoin parameters
+///
+///These parameters define how client wants to handle Payjoin.
+#[derive(Clone)]
+pub struct RequestBuilder(PdkRequestBuilder<'static>);
+
+impl From<PdkRequestBuilder<'static>> for RequestBuilder {
+    fn from(value: PdkRequestBuilder<'static>) -> Self {
+        Self(value)
+    }
+}
+
+impl RequestBuilder {
+    /// Prepare an HTTP request and request context to process the response
+    ///
+    /// An HTTP client will own the Request data while Context sticks around so
+    /// a `(Request, Context)` tuple is returned from `RequestBuilder::build()`
+    /// to keep them separated.
+    pub fn from_psbt_and_uri(psbt: String, uri: Arc<Uri>) -> Result<Self, PayjoinError> {
+        let psbt = payjoin::bitcoin::psbt::PartiallySignedTransaction::from_str(psbt.as_str())?;
+        match PdkRequestBuilder::from_psbt_and_uri(psbt, (*uri).clone().into()) {
+            Ok(e) => Ok(e.into()),
+            Err(e) => Err(e.into()),
+        }
+    }
```

Comment:
Where you don't have to call a block, there are lots of opportunities to be more succinct. Please don't take this too harshly, this is working code which matters more than anything else. Just suggestions for later iterations

```suggestion
    pub fn from_psbt_and_uri(psbt: String, uri: Arc<Uri>) -> Result<Self, PayjoinError> {
        let psbt = PartiallySignedTransaction::from_str(&psbt)?;
        PdkRequestBuilder::from_psbt_and_uri(psbt, (*uri).clone().into())
            .map(Into::into)
            .map_err(Into::into)
    }
```

Reply:
These suggestions are great. Your knowledge of the Rust Payjoin inner workings is going to help a lot in making this robust and efficient